### ISSUE TITLE
Regarding the community split

### ISSUE THREAD

**Issue by NickSeagull** [Wox-launcher/Wox#2960]
There has been an active fork going on at `jjw24/Wox`, and a discussion to split the Wox community, because the non-organization of this one. It ended up being renamed as a new project called [Flow](https://github.com/Flow-Launcher/)

@bao-qian stated that he has retaken the development again, and given that he has a bit more time now, perhaps it is a good time to organize the organization, add more members, and include changes from the fork.

Summoning @jjw24 , @theClueless , @kalaschnik , @SysC0mp , @JohnTheGr8 , @gdziedzic to hear their opinions on this.

We have already setup some infrastructure on Flow, like spectrum for managing questions, CI/CD, and other stuff. But given that *a lot* of people know Wox already, perhaps it'd make sense to merge them both again, and modernize the experience?

**Comment by bao-qian** [comment]
contribution is always welcome.
I added a few active contributor as members before, and I will add more members in the future.
I'm willing to add anyone active contribute to wox as member with no problem.

this repo is back to active development, with tons of bug fix and performance improvement.
You can try out these improvements by using new release.

**Comment by NickSeagull** [comment]
@jjw24 has done a lot of great work modernizing the UI and adding more features, perhaps he can be added among other from the Flow organization as main member of Wox one? Check out this:

![flow demo](https://user-images.githubusercontent.com/26427004/82151677-fa9c7100-989f-11ea-9143-81de60aaf07d.gif)

@jjw24 , would you be up for integrating your changes into this Wox?

**Comment by bao-qian** [comment]
the roadmap for recently release will focus on stability and performance (e.g. first query after launch is slow, although is faster now in new release).

the roadmap for new ui will be similar to below ui mock up.

![login_new](https://user-images.githubusercontent.com/1169912/82454291-2740ca80-9ae4-11ea-8408-74ca886b4ce3.gif)

and there are new theme to make it blur like windows.
![image](https://user-images.githubusercontent.com/1169912/82454754-bb129680-9ae4-11ea-83da-987f2d515a76.png)

**Comment by NickSeagull** [comment]
Would it be possible for me to join the GitHub organization in order to organize a bit better everything? I mean stuff like moving all questions into Spectrum rather than GitHub issues so roadmap is clear. Updating the README, etc....

Also, I'd help reviewing PRs, etc...

My key point is that Wox **is so great**, that we should be working all together in making it even better, as it is the go-to replacement of Alfred for Windows, and I'd love to make Wox be as high quality as Alfred

**Comment by bao-qian** [comment]
@NickSeagull 
invitation sent.

**Comment by JohnTheGr8** [comment]
Hi all. Refreshing to see this discussion started... adding a few thoughts and opinions to what has been said so far.

> @bao-qian stated that he has retaken the development again, and given that he has a bit more time now, perhaps it is a good time to organize the organization, add more members, and include changes from the fork.
> 
> Summoning @jjw24 , @theClueless , @Kalaschnik , @SysC0mp , @JohnTheGr8 , @gdziedzic to hear their opinions on this.

I think there is room for both Wox and Flow to coexist and thrive. Wox has a lot of traction already and can continue its development/maintenance, while Flow can introduce big changes (some have already been introduced) before the initial release. Flow can divert enough from Wox to justify its existence.

Of course, this is a community decision... If Flow contributors would prefer to focus their efforts on Wox again, that's great.

That said... @bao-qian I'm also happy to contribute to your effort here, by backporting changes (where it makes sense to), reviewing PRs, etc

> I mean stuff like moving all questions into Spectrum rather than GitHub issues so roadmap is clear. 

Please consider taking a vote/feedback from the community on this. Sorry for being negative, but Spectrum has been a bad experience for me: pages take too long to load, email notifications are no longer coming through, and the platform does not seem very flexible/powerful in general...

That said, a solution that takes questions/etc out of github issues is welcome. I think a platform for direct team/public chat is important, as well...

**Comment by NickSeagull** [comment]
Yep, I'll create a poll for this. Spectrum is my go-to solution, because it is the free one that works at least on average well. My biggest concern is having 500+ issues where most of them are general usage questions. These should be addressed separately in another place, in order to foster progress for the project :)

**Comment by jjw24** [comment]
Tools like Flow, Wox, PowerToys Run at their core are great because they are open for extensions by community made plugins. I think that's the most important, there is definitely room for all three to co-exist. My perspective, we could start the discussion including Microsoft to have the same plugin interface. This will mean plugin developers when they develop, it's for all three launchers, and plugins are shared and transferable.

I would say, just for Flow and Wox, they don't need to have the same set of changes, but for their progress if both project's contributors are up for it, they can help with both, but more importantly these projects (hopefully PT Run as well) should be using the same plugin interface so the focus is more on helping plugin community to thrive.

**Comment by JohnTheGr8** [comment]
@jjw24 I think we should try to strike a balance between:

- making it simple for plugin developers to port their plugins from one launcher project to the other without a ton of effort.
- not hesitating to introduce changes to the plugin interface that would make the life of plugin developers easier

**Comment by bao-qian** [comment]
since there are two forks now, and we do have disagreement in terms of ui. 
so I think a clean code base should be introduced to encourage code sharing and feature port. 

- `Wox.WPF` for current wpf releated ui code
- `Wox.WinUI` for next gen ui to align with ms latest winui library.
- `Wox.Core` for core functionality, like plugin management.
- `Wox.Plugin` for plugin developer.
- `Plugins/*` for built in plugins.

MS and Flow can built on top of Wox by add `Wox.Powertoy` / `Wox.Flow` in their repo, and share code with Wox by reusing `Wox.Core`, `Wox.Plugin` and most of plugins in `Plugins/*`.

**Comment by bao-qian** [comment]
@JohnTheGr8 
invitation sent.

**Comment by bao-qian** [comment]
in terms of features in Flow, every feature and bug fix has already been ported to wox from flow.
except for two, 1 .net core ugprade. 2. new ui .

**Comment by skjerns** [comment]
@bao-qian @jjw24 I think it would be really a shame to split the project. Is there no way you guys can align the roadmap? Don't let this be another open-office / libre-office. I don't think the goal and feature set diverges so much that it requires a fork and a new project (unlike e.g. opera/vivaldi or openelec/libreelec). 

Collaboration is always better! Unless there are good reasons for Flow to split if I'd strongly favor merging the projects again. I think until now most changes proposed by Flow could also be a setting?

**Comment by NickSeagull** [comment]
Agree with @skjerns

**Comment by bao-qian** [comment]
I'm also agree with @skjerns
I will do my best. I will merge bug fix / feature from forks, and also add more member to org.

**Comment by kalaschnik** [comment]
Just adding my two cents, as I was very inactive within the Flow community recently, and since @bao-qian is pushing on Wox again: **I would love to see merging Flow with Wox.** (as it seems to be already happendd), since it would be a shame to split the creative minds behind both projects.

Foremost, if we are all back in one repository we don’t need to worry about plugin compatibility. In any way Wox should be compatible with PowerToys Run Plugins, once they announced their specifications—which hopefully rely on Wox (pointing to https://github.com/microsoft/PowerToys/issues/3200). This would be crucial, as I think PowerToys Run will have the highest impact regarding its reach in the future.

Second, imo I really like the UI of the new PT Run / Wox concept. In anyway, I think if there are future fundamental UI disagreements, I would suggest to solve this by polling the users within the Wox repo. Yet, in general, I think it is good be close to the UI of PT Run, so people who switch from PT Run to Wox (or vice versa) feel familiar.

These were my 0.02$

**Comment by R-Studio** [comment]
@bao-qian it's nice to hear you and the community will continue with this project, but why is the last "stable" release from 2018:
![image](https://user-images.githubusercontent.com/7498854/99367575-6cd72800-28ba-11eb-8eaf-7582f0b873f4.png)

**Comment by R-Studio** [comment]
Any news on this?

**Comment by wangyang1633** [comment]
What is the status of this? Is Wox Launcher still being developed or has Flow Launcher over taken it?

**Comment by skjerns** [comment]
It seems like the active development on Wox has been at a minimum. Flow however receives updates more frequently.

@jjw24 It really is a shame that you two could not put your efforts together. To this date I see literally no difference between Flow and Wox besides some superficial changes. I just can't see why you could not align? It is a lose-lose situation for all of us.

**Comment by kalaschnik** [comment]
Biggest win for Flow is that there is no requirement for Everything. Plus you get the ability to search within files. Both facts ain’t superficial.

**Comment by R-Studio** [comment]
@bao-qian any news? Please don't let this projekt die!

**Comment by wangyang1633** [comment]
@R-Studio Why not just use Flow Launcher? I have been using it.

**Comment by R-Studio** [comment]
@wangyang1633 You are right and Flow Launcher is really a nice alternativ!

**Comment by HaleShaw** [comment]
Wox is a so good software! Please don't let it die. Are you still working on it? @bao-qian
To be honest, I personally don't want it to be split. I prefer its purity and simplicity.
I would love to be able to do something about it, but I'm not very good at C#. Maybe I will learn to develop in C# by myself in the future, and I hope I can help develop it in the future.

Although @jjw24 has done a lot of development on Flow, made a lot of upgrades to it. But I still want both of you to work together to make Wox better. I know it's very very hard, almost impossible.

**Comment by NickSeagull** [comment]
@HaleShaw Flow is literally Wox with fixes and additions

**Comment by HaleShaw** [comment]
> @HaleShaw Flow is literally Wox with fixes and additions

Yes, of cause. I know that Flow has many fixes and upgrades. Just like I want Wox to get better, I think so does @jjw24. And he proved himself by his actions, fixed Wox himself and created Flow.
But maybe I have some different ideas about UI as @bao-qian. I also hope the UI of Flow will be better in the future.
So I just said that I prefer Wox's purity and simplicity.

**Anyway, I just hope that someone will continue to develop and improve this software.**😊

**Comment by TheBestPessimist** [comment]
bad bot

**Comment by TheBestPessimist** [comment]
Suggestion to add the label `keep-fresh`

**Comment by qianlifeng** [comment]
Many years have passed, and Flow Launcher has developed quite well. Now the architectures of the two projects have diverged significantly, and I think it would be more appropriate to keep them as separate projects. Let's end this discussion here. Thank you all for your participation and contributions.
